packages/qwik/src/cli/check-client/get-latest-m-time.ts

```diff
@@ -0,0 +1,44 @@
+import path from 'path';
+import fs from 'fs/promises';
+
+/**
+ * Recursively finds the latest modification time (mtime) of any file in the given directory.
+ *
+ * @param {string} directoryPath - The directory path to search.
+ * @returns {Promise<number>} Returns the latest mtime (Unix timestamp in milliseconds), or 0 if the
+ *   directory doesn't exist or is empty.
+ */
+export async function getLatestMtime(directoryPath: string, timestamp: number): Promise<boolean> {
+  let returnValue = false;
+  async function traverse(dir: string): Promise<void> {
+    let items: Array<import('fs').Dirent>;
+    try {
+      items = await fs.readdir(dir, { withFileTypes: true });
+    } catch (err: any) {
+      if (err.code !== 'ENOENT') {
+        console.warn(`Cannot read directory ${dir}: ${err.message}`);
+      }
+      return;
+    }
+
+    for (const item of items) {
+      const fullPath = path.join(dir, item.name);
```

Comment:
Add `if (returnValue) { return }` before this line so you stop traversing when you found a newer file